src/ngx_http_lua_kong_log.c

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright 2019-2022 Kong Inc.
+
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+
+ *    http://www.apache.org/licenses/LICENSE-2.0
+
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include "ngx_http_lua_kong_common.h"
+
+
+int
+ngx_http_lua_kong_ffi_set_log_level(ngx_http_request_t *r, int level)
+{
+    r->connection->log->log_level = (ngx_uint_t) level;
+    r->connection->listening->log.log_level = (ngx_uint_t) level;
+    r->connection->listening->logp->log_level = (ngx_uint_t) level;
+    return NGX_OK;
+}
```

Comment:
@attenuation I added this test: https://github.com/Kong/lua-kong-nginx-module/blob/feat/FT-2818-dynamic_log_levels/t/008-log.t#L257-L284

Reply:
No, I mean if this change will be work in the current and new connection (newer than the current connection), because the new timer-ng library will allocate some timer before and reuse it, these timers maybe old than this change, Maybe it's not work for these timer. this test just test change timer context log, you can try to start timer in one connection(use sleep to wait another connection set), and change log_level in another connection

Reply:
```
    error_log logs/error.log notice;
    location = /test {
        content_by_lua_block {
            ngx.timer.at(0, function()
                ngx.sleep(10)
                ngx.log(ngx.DEBUG, "timer debug log")
            end)
            ngx.log(ngx.DEBUG, "content debug log")
        }
    }


    location = /test1 {
        error_log logs/error.log notice;
        content_by_lua_block {
            local log = require("resty.kong.log")
            ngx.log(ngx.NOTICE, "timer notice log")
            log.set_log_level(ngx.DEBUG)
            ngx.log(ngx.DEBUG, "timer debug log")
        }
    }
```

```
curl localhost/test
curl localhost/test1
```
Maybe you can see timer log_level is not change.

Reply:
And you just change the current listen log, Maybe it's not working for other listen 
```
 server {
    listen       80;
    location = /test {
        content_by_lua_block {
            local log = require("resty.kong.log")
            ngx.log(ngx.NOTICE, "timer notice log")
            log.set_log_level(ngx.DEBUG)
            ngx.log(ngx.DEBUG, "timer debug log")
        }
    }
}
 server {
    listen       81;
    location = /test {
        content_by_lua_block {
            ngx.timer.at(0, function()
                while( true )
                do
                    ngx.sleep(1)
                    ngx.log(ngx.DEBUG, "timer debug log") // maybe not working
                end
            end)
        }
    }
}
```

Reply:
@attenuation could you please re-check? now we're changing each listener log level, both for current and new requests. I have also added these two tests: https://github.com/Kong/lua-kong-nginx-module/blob/feat/FT-2818-dynamic_log_levels/t/008-log.t#L304-L409

Reply:
Maybe we can describle these limitation in doc? For old timer, i have a idea to slove this problem, we can check `cycle->new_log->log_level` in timer-ng callback, and then dynamic change old timer log_level.

Reply:
@attenuation we can already change old timer log_level: https://github.com/Kong/lua-kong-nginx-module/blob/feat/FT-2818-dynamic_log_levels/t/008-log.t#L587-L610